firstvoices/backend/views/sites_views.py

```diff
@@ -72,6 +74,13 @@ class SiteViewSet(FVPermissionViewSetMixin, ModelViewSet):
     pagination_class = None
     serializer_class = SiteDetailWriteSerializer
 
+    def initial(self, *args, **kwargs):
+        if self.action == "retrieve":
+            obj = self.get_object()
+            if obj.is_hidden and not can_view_hidden_site(self.request.user, obj):
```

Comment:
Rather than defining this here, we should add this predicate to the permission rule that is defined on the site model. Otherwise we are going to get confused about what permission is being applied where